typeform/typeform.py

```diff
@@ -0,0 +1,183 @@
+import time
+import copy
+import json
+import panoply
+import urllib2
+import datetime
+
+
+FETCH_LIMIT = 500
+DESTINATION = 'typeform'
+DESTINATION_POSTFIX = '_{__table}'
+BASE_URL = 'https://api.typeform.com/v1'
+DATE_PARSER_FORMAT = '%Y-%m-%dT%H:%M:%S'
+
+
+class Typeform(panoply.DataSource):
+
+    # constructor
+    def __init__(self, source, options):
+        super(Typeform, self).__init__(source, options)
+
+        if 'destination' not in source:
+            source['destination'] = DESTINATION
+
+        # append the destination postfix, which represent the form name
+        source['destination'] += DESTINATION_POSTFIX
+
+        # configure incremental pulling
+        self._incval = None
+        if (source.get('inckey')):
+            incval = source.get('incval')
+            self._incval = (incval if incval else
+                source.get('lastTimeSucceed') or source.get('lastRuntime'))
+
+            # validate that's the incremental value is a valid
+            # date that could be parsed to a Unix timestamp before
+            # including it in the process
+            self._incval = getTimestamp(self._incval)
+
+
+        forms = source.get('forms')
+        forms = forms if forms else []
+
+        # add an 'offset' attribute used for pagination for each
+        # different form, used as the actual number to use as offset
+        self._forms = map(lambda form: {
+            'id': form['id'],
+            'name': form['name'],
+            'offset': 0
+        }, forms)
+
+        self._key = source.get('key')
+        self._total = len(self._forms)
+
+    def read(self, n = None):
+        if len(self._forms) == 0:
+            # no more data to consume
+            return None
+
+        form = self._forms[0]
+        url = '%s/form/%s?key=%s&completed=true&offset=%s&limit=%s' % (
```

Comment:
Maybe this should be sent to a instance method that accepts a `form` and returns the `url` instead of having this manual url building in the way of the main code

This includes the following `if self._incval` part